internal/helm/controller/reconcile.go

```diff
@@ -225,6 +225,21 @@ func (r HelmOperatorReconciler) Reconcile(request reconcile.Request) (reconcile.
 		}
 		err = r.updateResourceStatus(o, status)
 		return reconcile.Result{RequeueAfter: r.ReconcilePeriod}, err
+	} else if !manager.IsUpgradeRequired() &&
+		!status.ContainsCondition(types.ConditionDeployed) &&
+		!status.ContainsCondition(types.ConditionReleaseFailed) {
+		status.SetCondition(types.HelmAppCondition{
+			Type:    types.ConditionDeployed,
+			Status:  types.StatusTrue,
+			Reason:  types.ReasonInstallSuccessful,
+			Message: manager.ReleaseMessage(),
+		})
```

Comment:
Can you move this to line ~334? We should just _always_ set the condition.

At that point, we can also look at `expectedRelease`, so I don't think we need the extra manager functions.

Lastly, we should look at the release version to know whether to set `ReasonInstallSuccessful` or `ReasonUpgradeSuccessful`.

WDYT? Will the problem be still be solved if those tweaks are made?